evaluation/utils/shared.py

```diff
@@ -181,34 +181,43 @@ def prepare_dataset(
     output_file: str,
     eval_n_limit: int,
     eval_ids: list[str] | None = None,
+    start_id: int | None = None,
 ):
     assert (
         'instance_id' in dataset.columns
     ), "Expected 'instance_id' column in the dataset. You should define your own unique identifier for each instance and use it as the 'instance_id' column."
     id_column = 'instance_id'
     logger.info(f'Writing evaluation output to {output_file}')
-    finished_ids = set()
+    finished_ids: set[int] = set()
```

Comment:
Actually, the ID is not strictly INT - in fact, they are mostly string across all the evals, which makes start_id harder to support?

Reply:
Oops, I'll change it to str then of course.

Reply:
Fixed